model/DataObject.php

```diff
@@ -1417,9 +1417,9 @@ public function getComponent($componentName) {
 		if($class = $this->has_one($componentName)) {
 			$joinField = $componentName . 'ID';
 			$joinID    = $this->getField($joinField);
-			
+
 			if($joinID) {
-				$component = $this->model->$class->byID($joinID);
+				$component = DataObject::get_one($class, 'ID = ' . $joinID);
```

Comment:
This is actually producing non-standard SQL. You need to quote your SQL identifiers if possible.

Also, when querying on ID there is the risk of an ambiguous field error (if a multi-table class is joined on ID).

If you use `DataObject::get_by_id($class, $joinID)` this is taken care of and uses the same cache. This is why I recommended it earlier. :)

```php
/**
	 * Return the given element, searching by ID
	 *
	 * @param string $callerClass The class of the object to be returned
	 * @param int $id The id of the element
	 * @param boolean $cache See {@link get_one()}
	 *
	 * @return DataObject The element
	 */
	public static function get_by_id($callerClass, $id, $cache = true) {
		if(is_numeric($id)) {
			if(is_subclass_of($callerClass, 'DataObject')) {
				$baseClass = ClassInfo::baseDataClass($callerClass);
				return DataObject::get_one($callerClass,"\"$baseClass\".\"ID\" = $id", $cache);

				// This simpler code will be used by non-DataObject classes that implement DataObjectInterface
			} else {
				return DataObject::get_one($callerClass,"\"ID\" = $id", $cache);
			}
		} else {
			user_error("DataObject::get_by_id passed a non-numeric ID #$id", E_USER_WARNING);
		}
	}
```